src/app/services/variable.service.ts

```diff
@@ -249,10 +250,9 @@ export class VariableService {
     nodes.forEach((node: GbTreeNode) => {
       if (node) {
         const val = fields.length < 2 ? node[fields[0]] : (node[fields[0]] || {})[fields[1]];
-        if (
-          values.includes(val)
-          && !this.selectedVariablesTree.includes(node)) {
+        if (values.includes(val) && !this.selectedVariablesTree.includes(node)) {
           this.selectedVariablesTree.push(node);
```

Comment:
These two lines do: 1) select a node 2) select its ancestors recursively. Instead, the newly created function could select a node with its ancestors recursively.